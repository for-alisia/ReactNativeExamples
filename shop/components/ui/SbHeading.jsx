import React from 'react';
import { View, StyleSheet } from 'react-native';

// Components
import SbText from './SbText';

// Theme
import theme from '../../theme';

const SbHeading = (props) => {
  const { children } = props;
  return (
    <View style={styles.container}>
      <View style={styles.headingWrapper}>
        <SbText style={styles.heading}>{children}</SbText>
        <View style={styles.line}></View>
      </View>
      <SbText style={styles.shadowHeading}>{children}</SbText>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'relative',
    marginTop: theme.margin.l,
    marginBottom: theme.margin.l,
  },
  headingWrapper: {
    alignSelf: 'flex-start',
  },
  heading: {},
  line: {
    height: 2,
    backgroundColor: theme.colors.secondary,
  },
  // Transform to theme or some other units
  shadowHeading: {
    fontFamily: theme.fonts.monserratBold,
    color: theme.colors.primaryLight,
    textTransform: 'uppercase',
    opacity: 0.25,
    fontSize: 40,
    position: 'absolute',
    top: -28,
    left: -6,
  },
});

export default SbHeading;
