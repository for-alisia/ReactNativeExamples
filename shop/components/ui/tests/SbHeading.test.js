// @ts-nocheck
// Dependencies
import React from 'react';
import { render } from '@testing-library/react-native';

// Component to test
import SbHeading from '../SbHeading';

describe('SbHeading UI Component', () => {
  const testText = 'Hello';

  it('renders a heading component with correct text', () => {
    const { getAllByText, toJSON } = render(<SbHeading>{testText}</SbHeading>);
    const foundText = getAllByText(testText);
    expect(foundText.length).toEqual(2);
    expect(foundText[0].props.children).toEqual(testText);
    expect(toJSON()).toMatchSnapshot();
  });
});
