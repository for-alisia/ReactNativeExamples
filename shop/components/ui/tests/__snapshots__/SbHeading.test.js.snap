// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SbHeading UI Component renders a heading component with correct text 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
      "marginTop": 16,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "flex-start",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3C3735",
          "fontFamily": "montserrat-regular",
          "fontSize": 16,
        }
      }
    >
      Hello
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#D8C496",
          "height": 2,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#ECBDBC",
        "fontFamily": "montserrat-bold",
        "fontSize": 40,
        "left": -6,
        "opacity": 0.25,
        "position": "absolute",
        "textTransform": "uppercase",
        "top": -28,
      }
    }
  >
    Hello
  </Text>
</View>
`;
